app/controllers/openstax/api/v1/api_controller.rb

```diff
@@ -27,17 +27,17 @@ class ApiController < ActionController::Base
         before_filter :doorkeeper_authorize!, if: :token_user?
 
         # Except for users logged in via a cookie, we can disable CSRF protection and enable CORS
-        skip_before_filter :verify_authenticity_token, unless: :session_user?
+        skip_before_filter :verify_authenticity_token, unless: :local_session_user?
+        skip_before_filter :authenticate_user!, only: :options
         skip_before_filter :verify_authenticity_token, only: :options
-        before_filter :set_cors_preflight_headers, only: :options
-        before_filter :set_cors_headers
-        after_filter :set_cors_headers
+
+        before_filter  :maybe_set_cors_headers
 
         # Keep old current_user method so we can use it
         alias_method :current_session_user, OpenStax::Api.configuration.current_user_method
 
         # Ensure we will never again confuse human users and api users
-        undef_method OpenStax::Api.configuration.current_user_method
+        # undef_method OpenStax::Api.configuration.current_user_method
 
```

Comment:
@jpslav this line is removing the `current_user` from Accounts `SignInState` https://github.com/openstax/accounts/blob/master/lib/sign_in_state.rb which causes failures in various controllers there.  Any thoughts on  the best way to proceed?  Should account's controllers be modified to no longer call `current_user`, or should the initializer in accounts set `current_user_method` to something else?